src/Symfony/Component/JsonPath/JsonCrawler.php

```diff
@@ -618,4 +1070,39 @@ private function transformJsonPathRegex(string $pattern): string
 
         return $result;
     }
+
+    /**
+     * A value is considered a JSON array structure if:
+     * - It's a list, just like in PHP
+     * - For single element arrays, it must be determined if it was originally a JSON array or object
+     *   per RFC 9535 section 4.5: numeric index selectors only work on arrays, not objects
+     */
+    private function isJsonArrayStructure(mixed $value): bool
+    {
+        return \is_array($value) && array_is_list($value) && (1 !== \count($value) || !isset($value[0]));
```

Comment:
why wouldn't `['foo']` be a JSON array structure ?

Reply:
It seems really not that easy to fix... Just in case, @mtarld, is there something in JsonStreamer that could also help here? We would need to "easily" differentiate an array from an object with numeric indices. Maybe `Read/Lexer.php`?

Reply:
For the case of strings using `json_decode`, we might decode objects into `stdClass` to distinguish objects from arrays. But this suffers from other limitations as this won't support object keys that are not valid names of PHP properties.